R/gamlss-tidiers.R

```diff
@@ -1,39 +1,53 @@
-#' @templateVar class gamlss
-#' @template title_desc_tidy
-#'
-#' @param x A `gamlss` object returned from [gamlss::gamlss()].
-#' @template param_quick
-#' @template param_unused_dots
-#' 
-#' @evalRd return_tidy(
-#'   parameter = "Type of coefficient being estimated: `mu`, `sigma`,
-#'     `nu`, or `tau`.",
-#'   "term",
-#'   "estimate",
-#'   "std.error",
-#'   "statistic",
-#'   "p.value"
-#' )
-#' 
+#' Tidying methods for gamlss objects
+#'
+#' Tidying methods for "gamlss" objects from the gamlss package.
+#'
+#' @param x A "gamlss" object
+#' @param quick Whether to perform a fast version, and return only the coefficients
+#' @param conf.int whether to return confidence intervals
+#' @param ... arguments passed to \code{confint.gamlss}
+#'
+#' @name gamlss_tidiers
+#'
+#' @template boilerplate
+#'
+#' @return A tibble with one row for each coefficient, containing columns:
+#'   \item{parameter}{type of coefficient being estimated: \code{mu}, \code{sigma}, \code{nu}, or \code{tau}}
+#'   \item{term}{term in the model being estimated and tested}
+#'   \item{estimate}{estimated coefficient}
+#'   \item{std.error}{standard error}
+#'   \item{statistic}{t-statistic}
+#'   \item{p.value}{two-sided p-value}
+#'
 #' @examples
-#' 
-#' library(gamlss)
-#' 
-#' g <- gamlss(
-#'   y ~ pb(x),
-#'   sigma.fo = ~ pb(x),
-#'   family = BCT,
-#'   data = abdom,
-#'   method = mixed(1, 20)
-#' )
-#'
-#' tidy(g)
+#' \dontrun{
+#' if (requireNamespace("gamlss", quietly = TRUE) &&
```

Comment:
The examples as well -- let's keep what was in `broom`. The only new thing here is checking if `gamlss` is loaded. Technically you should do this, but it makes the examples harder to read and is never the case in practice, so the standard in `broom` is not to do this.